File: client/app/src/commonMain/kotlin/com/jahirtrap/cconnect/ui/theme/Accents.kt
package com.jahirtrap.cconnect.ui.theme

import androidx.compose.ui.graphics.Color

// Material 500 palette.
val ACCENTS: List<Pair<String, Color>> = listOf(
    "Red" to Color(0xFFF44336),
    "Pink" to Color(0xFFE91E63),
    "Purple" to Color(0xFF9C27B0),
    "Deep Purple" to Color(0xFF673AB7),
    "Indigo" to Color(0xFF3F51B5),
    "Blue" to Color(0xFF2196F3),
    "Light Blue" to Color(0xFF03A9F4),
    "Cyan" to Color(0xFF00BCD4),
    "Teal" to Color(0xFF009688),
    "Green" to Color(0xFF4CAF50),
    "Light Green" to Color(0xFF8BC34A),
    "Lime" to Color(0xFFCDDC39),
    "Yellow" to Color(0xFFFFEB3B),
    "Amber" to Color(0xFFFFC107),
    "Orange" to Color(0xFFFF9800),
    "Deep Orange" to Color(0xFFFF5722),
    "Brown" to Color(0xFF795548),
    "Grey" to Color(0xFF9E9E9E),
    "Blue Grey" to Color(0xFF607D8B),
)

// Stored instead of a palette index when the accent should follow the system colour.
const val DYNAMIC_ACCENT = -1

fun accentAt(index: Int): Color = ACCENTS.getOrElse(index) { ACCENTS[4] }.second
fun accentNameAt(index: Int): String? = ACCENTS.getOrNull(index)?.first

File: tauri/src/lib/design/accents.ts
export const ACCENTS: ReadonlyArray<{ name: string; value: string }> = [
  { name: "Red", value: "#f44336" },
  { name: "Pink", value: "#e91e63" },
  { name: "Purple", value: "#9c27b0" },
  { name: "Deep Purple", value: "#673ab7" },
  { name: "Indigo", value: "#3f51b5" },
  { name: "Blue", value: "#2196f3" },
  { name: "Light Blue", value: "#03a9f4" },
  { name: "Cyan", value: "#00bcd4" },
  { name: "Teal", value: "#009688" },
  { name: "Green", value: "#4caf50" },
  { name: "Light Green", value: "#8bc34a" },
  { name: "Lime", value: "#cddc39" },
  { name: "Yellow", value: "#ffeb3b" },
  { name: "Amber", value: "#ffc107" },
  { name: "Orange", value: "#ff9800" },
  { name: "Deep Orange", value: "#ff5722" },
  { name: "Brown", value: "#795548" },
  { name: "Grey", value: "#9e9e9e" },
  { name: "Blue Grey", value: "#607d8b" },
];

export const DEFAULT_ACCENT_INDEX = 4;

export const DYNAMIC_ACCENT = -1;

export const accentAt = (index: number): string =>
  (ACCENTS[index] ?? ACCENTS[DEFAULT_ACCENT_INDEX]).value;

export const accentNameAt = (index: number): string | undefined => ACCENTS[index]?.name;
